src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -164,17 +169,102 @@ impl ConnectionOptionExtracted {
+                        Some(AwsAuth::Credentials(AwsCredentials {
+                            access_key_id,
+                            secret_access_key: secret_access_key.into(),
+                            session_token,
+                        }))
+                    }
+                    (None, Some(_), _) => {
+                        return Err(PlanError::MissingRequiredOptions {
+                            option_names: vec![ConnectionOptionName::AccessKeyId.to_string()],
+                            item_type: CatalogItemType::Connection,
+                            item_sub_type: Some("AWS Credentials".to_string()),
+                        })
+                    }
+                    (Some(_), None, _) => {
+                        return Err(PlanError::MissingRequiredOptions {
+                            option_names: vec![ConnectionOptionName::SecretAccessKey.to_string()],
+                            item_type: CatalogItemType::Connection,
+                            item_sub_type: Some("AWS Credentials".to_string()),
+                        })
+                    }
+                    (None, None, Some(_)) => {
+                        return Err(PlanError::MissingRequiredOptions {
+                            option_names: vec![
+                                ConnectionOptionName::AccessKeyId.to_string(),
+                                ConnectionOptionName::SecretAccessKey.to_string(),
+                            ],
+                            item_type: CatalogItemType::Connection,
+                            item_sub_type: Some("AWS Credentials".to_string()),
+                        })
+                    }
+                    _ => None,
+                };
+
+                let assume_role = match (self.assume_role_arn, self.assume_role_session_name) {
+                    (Some(arn), session_name) => {
+                        let mz_arn =
+                            scx.catalog.aws_external_connection_role().ok_or_else(|| {
+                                PlanError::Catalog(CatalogError::MissingState(
+                                    "AWS external connection role".to_string(),
+                                ))
+                            })?;
+                        let external_id_prefix =
+                            scx.catalog.aws_external_id_prefix().ok_or_else(|| {
+                                PlanError::Catalog(CatalogError::MissingState(
+                                    "AWS External ID prefix".to_string(),
+                                ))
+                            })?;
+                        Some(AwsAuth::AssumeRole(AwsAssumeRole {
+                            arn,
+                            session_name,
+                            mz_principal: MaterializePrincipal {
+                                arn: mz_arn,
+                                external_id_prefix,
+                            },
+                        }))
```

Comment:
See future comment about not storing `mz_principal` in the connection config; if you do so you shouldn't need to pull anything out of the catalog or add the new `MissingState` variant.

Reply:
updated